src/utils/resolveXYScales.js

```diff
@@ -53,85 +55,82 @@ function hasAllMargins(margin) {
   const marginKeys = ['top', 'bottom', 'left', 'right'];
   return _.isObject(margin) && _.every(marginKeys, k => _.has(margin, k));
 }
-
-function hasGoodXY(obj, isValid = (v => !_.isUndefined(v))) {
-  return _.isObject(obj) && isValid(obj.x)
+function hasAllSpacing(spacing) {
+  const spacingKeys = ['top', 'bottom', 'left', 'right'];
+  return _.isObject(spacing) && _.every(spacingKeys, k => _.has(spacing, k));
 }
-
-function mapStaticOnChildren(children, methodName, passProps = {}) {
+function runFunctionOnChildren(children, passedFunction, ...passProps) {
   // returns the result of looping over all children and calling a static method on each one
```

Comment:
I think this function needs some clarification.
* It's not just running a function (like `forEach`), it's returning all the results (like `map`). `mapOverChildren` or similar seems a better name
* The comment is incorrect. This function does not call any static methods.
* I don't really understand what's happening with the `passProps`, and why strings are resolved to `child[item]` while non-strings are passed through? Some clarifying comments would be appreciated.